### PR TITLE
refactor(RegExp): use /.../ to create RegExp literal

### PR DESCRIPTION
**Pull request by vicb**
[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2695)

### CODE REVIEW

**Comment by vicb** [comment]
@pkozlowski-opensource I'll squash the other PR here

File: modules/angular2/src/render/dom/shadow_dom/shadow_css.ts [hunk 1]
```diff
@@ -488,12 +488,15 @@ export class ShadowCss {
   }
 }
 
-var _cssContentNextSelectorRe = RegExpWrapper.create(
-    'polyfill-next-selector[^}]*content:[\\s]*?[\'"](.*?)[\'"][;\\s]*}([^{]*?){', 'im');
+var _cssContentNextSelectorRe =
+    /polyfill-next-selector[^
```
Comment (vicb) [inline modules/angular2/src/render/dom/shadow_dom/shadow_css.ts]: @mprobst clang oops !

File: modules/angular2/src/render/dom/shadow_dom/shadow_css.ts [hunk 2]
```diff
@@ -488,12 +488,15 @@ export class ShadowCss {
...
+}]*content:[\s]*?['"](.*?)['"][;\s]*}([^{]*?){/gim;
 var _cssContentRuleRe =
-    RegExpWrapper.create('(polyfill-rule)[^}]*(content:[\\s]*[\'"](.*?)[\'"])[;\\s]*[^}]*}', 'im');
-var _cssContentUnscopedRuleRe = RegExpWrapper.create(
-    '(polyfill-unscoped-rule)[^}]*(content:[\\s]*[\'"](.*?)[\'"])[;\\s]*[^}]*}', 'im');
+    /(polyfill-rule)[^
```
Comment (vicb) [inline modules/angular2/src/render/dom/shadow_dom/shadow_css.ts]: clang

File: modules/angular2/src/render/dom/shadow_dom/shadow_css.ts [hunk 3]
```diff
@@ -488,12 +488,15 @@ export class ShadowCss {
...
+}]*(content:[\s]*['"](.*?)['"])[;\s]*[^}]*}/gim;
+var _cssContentUnscopedRuleRe =
+    /(polyfill-unscoped-rule)[^
```
Comment (vicb) [inline modules/angular2/src/render/dom/shadow_dom/shadow_css.ts]: clang

File: modules/angular2/test/change_detection/parser/unparser.ts [hunk 4]
```diff
@@ -23,12 +23,11 @@ import {
...
 
 
-import {StringWrapper, RegExpWrapper, isPresent, isString} from 'angular2/src/facade/lang';
-
-var quoteRegExp = RegExpWrapper.create('"');
+import {StringWrapper, isPresent, isString} from 'angular2/src/facade/lang';
 
 export class Unparser implements AstVisitor {
-  private _expression: string;
+  private static _quoteRegExp = / "/g;
```
Comment (vicb) [inline modules/angular2/test/change_detection/parser/unparser.ts]: clang

**Comment by vicb** [comment]
blocked on https://github.com/angular/ts2dart/pull/219

**Comment by tbosch** [comment]
@vicb is this still blocked as the issue in ts2dart is merged?

**Comment by vicb** [comment]
@Tobias rebased - yet to be tested on Travis (the build is broken)

_Note: there is still an issue with clang-format it explains why it has been disabled on some sections_

**Comment by tbosch** [comment]
Try to make your lines shorter. Clang-format has a bug when the line is
close to the 120chars limit...
Otherwise try to turn off clang-format, see element_injector_spec.ts on the
top

**Comment by vicb** [comment]
blocked on https://github.com/angular/ts2dart/pull/235

**Comment by vicb** [comment]
@mprobst ptal

File: modules/angular2/src/render/api.ts [hunk 5]
```diff
@@ -156,6 +151,10 @@ export class DirectiveMetadata {
   hostProperties: Map<string, string>;
   hostAttributes: Map<string, string>;
   hostActions: Map<string, string>;
+  // group 1: "property" from "[property]"
+  // group 2: "event" from "(event)"
+  // group 3: "action" from "@action"
+  private static _hostRegExp = /^(?:(?:\[([^\]]+)\])|(?:\(([^\)]+)\))|(?:@(.+)))$/g;
```
Comment (mprobst) [inline modules/angular2/src/render/api.ts]: FYI I think within modules, there really isn't much of a need for static fields, top level variables are entirely fine and require less syntactic bruhaha. But whatever you like better is fine.

**Comment by vicb** [comment]
landed as 447926dc083c1c77cef290d27daf45f86e815863

**Pull request closed without merging**